### PR TITLE
feat: add support for FUSE connections

### PR DESCRIPTION
**Pull request by enocom**
This commit also ensures that closing the proxy.Client blocks until all
listeners are closed.

### CODE REVIEW

File: internal/proxy/fuse_test.go [hunk 1]
```diff
@@ -110,13 +101,212 @@ func TestREADME(t *testing.T) {
...
+			fuseTempDir:  fuseTempDir,
+		},
+		{
+			desc:         "postgres connections create a directory with a special file",
+			wantInstance: "proj:region:pg",
+			socketPath:   filepath.Join(fuseDir, "proj:region:pg", ".s.PGSQL.5432"),
+			fuseTempDir:  fuseTempDir,
+		},
+		{
+			desc:         "connecting creates intermediate temp directories",
```
Comment (kurtisvg) [inline internal/proxy/fuse_test.go:153]: Should we have some failures cases in here?

What if we dial an instance that doesn't exist? what happens?

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -34,6 +36,44 @@ import (
 	"golang.org/x/oauth2"
 )
 
+var (
+	// Instance connection name is the format <PROJECT>:<REGION>:<INSTANCE>
+	// Additionally, we have to support legacy "domain-scoped" projects (e.g. "google.com:PROJECT")
+	connNameRegex = regexp.MustCompile("([^:]+(:[^:]+)?):([^:]+):([^:]+)")
```
Comment (kurtisvg) [inline internal/proxy/proxy.go:42]: Should we consider just exposing these from the Go connector?

**Review by kurtisvg — APPROVED** [review]

File: internal/proxy/fuse_test.go (hunk 1 above)
Comment (kurtisvg) [reply internal/proxy/fuse_test.go:153]: I found failure cases below, closing this.

File: internal/proxy/proxy.go (hunk 2 above)
Comment (enocom) [reply internal/proxy/proxy.go:42]: Yes, I'll do that.

**Pull request merged**